app/src/main/java/de/westnordost/streetcomplete/overlays/street_furniture/StreetFurnitureOverlayForm.kt

```diff
@@ -0,0 +1,159 @@
+package de.westnordost.streetcomplete.overlays.street_furniture
+
+import android.os.Bundle
+import android.view.View
+import androidx.core.content.ContentProviderCompat
+import de.westnordost.osmfeatures.Feature
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.edits.ElementEditAction
+import de.westnordost.streetcomplete.data.osm.edits.create.CreateNodeAction
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.UpdateElementTagsAction
+import de.westnordost.streetcomplete.data.osm.geometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+import de.westnordost.streetcomplete.data.osm.mapdata.Node
+import de.westnordost.streetcomplete.databinding.FragmentOverlayStreetFurnitureBinding
+import de.westnordost.streetcomplete.osm.IS_DISUSED_STREET_FURNITURE_EXPRESSION
+import de.westnordost.streetcomplete.osm.IS_STREET_FURNITURE_INCLUDING_DISUSED_EXPRESSION
+import de.westnordost.streetcomplete.overlays.AbstractOverlayForm
+import de.westnordost.streetcomplete.util.getLocalesForFeatureDictionary
+import de.westnordost.streetcomplete.util.getLocationLabel
+import de.westnordost.streetcomplete.util.ktx.geometryType
+import de.westnordost.streetcomplete.util.ktx.viewLifecycleScope
+import de.westnordost.streetcomplete.view.controller.FeatureViewController
+import de.westnordost.streetcomplete.view.dialogs.SearchFeaturesDialog
+import kotlinx.coroutines.launch
+
+class StreetFurnitureOverlayForm : AbstractOverlayForm() {
+
+    override val contentLayoutResId = R.layout.fragment_overlay_street_furniture
+    private val binding by contentViewBinding(FragmentOverlayStreetFurnitureBinding::bind)
+
+    private var originalFeature: Feature? = null
+
+    private lateinit var featureCtrl: FeatureViewController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val element = element
+        originalFeature = element?.let {
+            if (IS_DISUSED_STREET_FURNITURE_EXPRESSION.matches(element)) {
+                DummyFeature(
+                    "street_furniture/unknown_disused",
+                    requireContext().getString(R.string.unknown_disused_street_furniture),
+                    "ic_preset_maki_marker_stroked",
+                    element.tags
+                )
```

Comment:
should be moved to the bottom, i.e.

```
originalFeature = featureDictionary.findTheFeature(...)
    ?: ( if (is disused) dummyDisused else dummy )
```
because it should be shown correctly when in the future, iD has some presets for that.

---

Actually... even better would be the following, as we can not expect that iD presets will ever contain (all the) disused features. Maybe then also the same for the shop overlay:

1. Copy the element tags and strip any disused prefix
2. Find the feature with these tags
3. If the feature matches the disused expression, wrap the returned feature in a `data class DisusedFeature(private var feature: Feature)` which changes the name etc. to e.g. "Bench (disused)"

And voila, we have disused features support.

Reply:
Heh, I just went here to write that I  plan to implement first part of this while waiting at airport.